commit 6370cc3bbd8a0f9bf975b013781243ab147876c6
Author: Aleksandr Nogikh <nogikh@google.com>
Date:   Thu Oct 29 17:36:19 2020 +0000

    net: add kcov handle to skb extensions
    
    Remote KCOV coverage collection enables coverage-guided fuzzing of the
    code that is not reachable during normal system call execution. It is
    especially helpful for fuzzing networking subsystems, where it is
    common to perform packet handling in separate work queues even for the
    packets that originated directly from the user space.
    
    Enable coverage-guided frame injection by adding kcov remote handle to
    skb extensions. Default initialization in __alloc_skb and
    __build_skb_around ensures that no socket buffer that was generated
    during a system call will be missed.
    
    Code that is of interest and that performs packet processing should be
    annotated with kcov_remote_start()/kcov_remote_stop().
    
    An alternative approach is to determine kcov_handle solely on the
    basis of the device/interface that received the specific socket
    buffer. However, in this case it would be impossible to distinguish
    between packets that originated during normal background network
    processes or were intentionally injected from the user space.
    
    Signed-off-by: Aleksandr Nogikh <nogikh@google.com>
    Acked-by: Willem de Bruijn <willemb@google.com>
    Signed-off-by: Jakub Kicinski <kuba@kernel.org>

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index d7a7bc3b6098..d171a032db78 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1870,6 +1870,7 @@ config KCOV
 	depends on CC_HAS_SANCOV_TRACE_PC || GCC_PLUGINS
 	select DEBUG_FS
 	select GCC_PLUGIN_SANCOV if !CC_HAS_SANCOV_TRACE_PC
+	select SKB_EXTENSIONS
 	help
 	  KCOV exposes kernel code coverage information in a form suitable
 	  for coverage-guided fuzzing (randomized testing).